### DISCUSSION TITLE
Enable zip compression for exported assets

### DISCUSSION THREAD

**Discussion by ganczarek** [apache/superset#27467] [Ideas]
[Superset uses](https://github.com/search?q=repo%3Aapache%2Fsuperset+language%3Apython+%2Fwith+ZipFile%28.*%22w%22%29%2F+AND+NOT+path%3Atest&type=code) default ZipFile compression, which is [ZIP_STORED](https://docs.python.org/3/library/zipfile.html#zipfile.ZipFile). Hence, exported assets are not compressed. I suspect it's not intentional or at least should be configurable. 

#### How to reproduce the bug

Export all assets with `/api/v1/assets/export/` endpoint. Other export endpoints - like `/api/v1/dashboard/export/` - also don't compress results.

### Expected results

Result zip file is compressed. 

### Actual results

Result zip file is not compressed.

### Environment

- browser type and version: n/a
- superset version: `2.1.0`
- python version: `3.8.13`
- node.js version: n/a
- any feature flags active: `VERSIONED_EXPORT: True` (enabled by deafult)

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [ ] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by rusackas** [comment]
Interesting find. CC @betodealmeida in case there are feelings/complications around this. 

I suspect that the non-compressed ZIP isn't causing an issue for most people (the files are small), and everything _works_, so this is not a bug. I'll move it to a discussion. 

I think the main reason NOT to compress things is that it would add another dependency for the compression module... and I'm not sure if people have feelings about which is best.

**Comment by rusackas** [comment]
Also, the PR for this seems pretty straightforward. Anyone is welcome to open that, and ake the discussion to the PR (rather than here)

**Comment by betodealmeida** [comment]
#TIL!

I don't think adding compression would hurt, specially if we use `zipfile.ZIP_DEFLATED`, given the note:

> The ZIP file format specification has included support for bzip2 compression since 2001, and for LZMA compression since 2006. However, some tools (including older Python releases) do not support these compression methods, and may either refuse to process the ZIP file altogether, or fail to extract individual files.

@rusackas maybe we should leave this open as an issue with the tag "good first issue"?
